plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/CachingJdbcClient.java

```diff
@@ -286,7 +288,26 @@ public PreparedStatement getPreparedStatement(Connection connection, String sql)
     @Override
     public TableStatistics getTableStatistics(ConnectorSession session, JdbcTableHandle handle, TupleDomain<ColumnHandle> tupleDomain)
     {
-        return delegate.getTableStatistics(session, handle, tupleDomain);
+        if (!handle.isNamedRelation()) {
+            return delegate.getTableStatistics(session, handle, tupleDomain);
```

Comment:
Why skip the cache here? 
There is no reason to discern cache logic between "predicate pushdown" vs "aggregation pushdown" cases, is there?

Reply:
I am not able to create a cache key for aggregation pushdow, while for predicate pushdown I cache constraint.

Reply:
Why can you not?

Reply:
Sorry, I should have mention that. It does not have a table name

Reply:
You don't have to put table name in the cache key, you can put JdbcRelationHandle instead

actually, here you're ignoring the TupleDomain embedded in the `handle` (in the cache key), which is not correct.

Reply:
I use handle in key now. But I still can cache only named relations as otherwise I am unable to invalidate cache without a table name.

Reply:
Good point. I am find to leave this as is, but we should at least have an issue for this -- we would collect source tables in the query-based handle, and use that for invalidation.

Or, simpler, we could have separate, shorter TTL for them, so there isn't a big deal in lack of invalidation.

In any case, we should have cache them for the duration of transaction.

Reply:
`io.trino.plugin.jdbc.JdbcQueryRelationHandle` is not designed to be cache key. Adding hashCode/equals and equals might be not a solution. As this could only work if we are lucky and query is repeated so get exactly same pushdowns.

Do you think it is possible that remote data source would be capable to provide stats for such table handle? I know only one database that provides stats for arbitrary query (I didn't do research, just out of head) :)

Reply:
> I know only one database that provides stats for arbitrary query

Continuing this thread. We might consider skip using using in cache keys pushdown predicate as that could also be not easily usable in getting stats from remote data source.

Reply:
> As this could only work if we are lucky and query is repeated so get exactly same pushdowns.

that's not unlikely:

- same query being repeated by the user
- stats being acquired more than once during query planning -- here caching within transaction would be sufficient, but it currently depends on the code here (as you are removing `TransactionCachingJdbcClient` class)